### DISCUSSION TITLE
Adding auto-completion

### DISCUSSION THREAD

**Discussion by TylerGrantSmith** [ropensci/targets#747] [Ideas]
I'm not sure if this is something you'd be interested in adding, but I've found it useful for my own purposes to implement [tab-completion in Rstudio](https://github.com/TylerGrantSmith/targets/tree/rstudio-completion)  for my targets objects.  This proof-of-concept implementation works with `tar_read` and `tar_load` using code adapted from paletteer (https://github.com/EmilHvitfeldt/paletteer/blob/master/R/completeme.R).  It works great!

Example of what it looks like:
![image](https://user-images.githubusercontent.com/39497067/148612241-c7273c6a-d761-4958-9dcc-c4249c6b45fb.png)

**Comment by wlandau** [comment]
Nice idea! I was wondering how autocompletion could be done. Would be nice to have autocompletion for the arguments of `tar_load()` and `tar_read()`. However, I am concerned about using ideas we get from reading GPL code. Both `palenteer` and `completeme` are GPL-3, so `targets` would need to switch from MIT to GPL-3, which is too high a cost. Also, I think both `targets` and the community would be better served with fully general-purpose function argument autocompletion functionality on CRAN. (Unfortunately, I see the last commit to `completeme` was April 2019, so maybe it would need to be revived or superseded somehow.)
